Added Military elite

todorzag opened the pull request.

todorzag committed:
Added MilitaryElite

todorzag committed:
Added Task Document

mvlt commented on MilitaryElite/MilitaryElite/LieutenantGeneral.cs:
```diff
@@ -0,0 +1,39 @@
+    {
+        public List<ISoldier> CommandingPrivates { get; }
+
+        public LieutenantGeneral
```
Shouldn't we be able to add privates rather than passing then to the constructor?

mvlt commented on MilitaryElite/MilitaryElite/Engineer.cs:
```diff
@@ -0,0 +1,40 @@
+
+namespace MilitaryElite
+{
+    internal class Engineer : SpecialisedSoldier
```
Missing interface for Engineer

mvlt commented on MilitaryElite/MilitaryElite/Engineer.cs:
```diff
@@ -0,0 +1,40 @@
+        public List<Repair> Repairs { get; }
+
+        public Engineer
+            (int id, string firstName, string lastName, decimal salary, string corps, List<Repair> repairs) 
```
Repairs should be added by the object not passed in the constructor

mvlt commented on MilitaryElite/MilitaryElite/Repair.cs:
```diff
@@ -0,0 +1,25 @@
+
+namespace MilitaryElite
+{
+    internal class Repair
```
Why internal class?
Why not using interface for it also?

mvlt reviewed (changes requested):
Use folders folder system to separate models from interfaces and others of some other logic
Use enums for different type of corps.
Add also Spy and Commando

mvlt commented on MilitaryElite/MilitaryElite/Program.cs:
```diff
@@ -0,0 +1,89 @@
+            Army.Add(new Engineer(id, firstName, lastName, salary, corp, repairs));
+        }
+
+        private static List<ISoldier> GetPrivatesFromInput(string[] privateIds)
```
Should be added by LieutenantGeneral

mvlt commented on MilitaryElite/MilitaryElite/Program.cs:
```diff
@@ -0,0 +1,89 @@
+        }
+
+        private static List<Repair> GetRepairsFromInput(string[] repairsInfo)
+        {
```
Should be added by Engineer

todorzag committed:
Added Commando, Spy and requested changes

mvlt commented on MilitaryElite/MilitaryElite/Other/Mission.cs:
```diff
@@ -0,0 +1,30 @@
+
+namespace MilitaryElite.Other
+{
+    internal class Mission
```
Missing interface

mvlt commented on MilitaryElite/MilitaryElite/Other/Repair.cs:
```diff
@@ -0,0 +1,25 @@
+
+namespace MilitaryElite.Other
+{
+    internal class Repair
```
Missing interface

mvlt reviewed (changes requested):
More appropriate will be to have a Models folder and if you want to separate Solder's models from other models - do that in Models folder, because now we have interfaces, other and soldiers at same level.

mvlt commented on MilitaryElite/MilitaryElite/Other/Mission.cs:
```diff
@@ -0,0 +1,30 @@
+
+        public void Complete()
+        {
+            State = "Finished";
```
Better to have enumeration for different mission states. What will happen if the state is already 'Finished'

mvlt commented on MilitaryElite/MilitaryElite/Soldiers/SpecialisedSoldier.cs:
```diff
@@ -0,0 +1,19 @@
+        public SpecialisedSoldier(int id, string firstName, string lastName, decimal salary, string corps)
+            : base(id, firstName, lastName, salary)
+        {
+            Corps = corps;
```
What will happen if I pass invalid corps?

mvlt commented on MilitaryElite/MilitaryElite/Program.cs:
```diff
@@ -0,0 +1,91 @@
+
+            while (input != "End")
+            {
+                string[] split = input.Split(" ");
```
split is not the best name

mvlt commented on MilitaryElite/MilitaryElite/Soldiers/LieutenantGeneral.cs:
```diff
@@ -0,0 +1,52 @@
+            CommandingPrivates = new List<ISoldier>();
+        }
+
+        public void AddPrivates(string[] privateIds)
```
This function should do only .Add

mvlt commented on MilitaryElite/MilitaryElite/Soldiers/Commando.cs:
```diff
@@ -0,0 +1,54 @@
+            Missions = new List<Mission>();
+        }
+
+        public void AddMissions(string[] missionsInfo)
```
same - AddMission

mvlt commented on MilitaryElite/MilitaryElite/Soldiers/Engineer.cs:
```diff
@@ -0,0 +1,56 @@
+            Repairs = new List<Repair>();
+        }
+
+        public void AddRepairs(string[] repairsInfo)
```
same

mvlt commented on MilitaryElite/MilitaryElite/Program.cs:
```diff
@@ -0,0 +1,91 @@
+                int id = int.Parse(split[1]);
+                string firstName = split[2];
+                string lastName = split[3];
+                decimal salary = decimal.Parse(split[4]);
```
Does all of the soldiers have salary?

mvlt commented on MilitaryElite/MilitaryElite/Program.cs:
```diff
@@ -0,0 +1,91 @@
+                decimal salary = decimal.Parse(split[4]);
+
+
+                string[] secondaryInformation = split[5..split.Length];
```
Bad naming. Do all of them have 5th element?

todorzag committed:
Changes Made:
 - Added Interfaces for Repair and Mission
 - Remade Add logic for LieutenantGeneral, Engineer and Commando
 - Added Enumerable States for Mission
 - Renamed variables in Program

todorzag committed:
Small changes

todorzag committed:
Rearranged files

todorzag committed:
Rearranged folders

todorzag committed:
Renamed folders
